### PR TITLE
make CompileStep.addRPath preserve relative paths

### LINKED ISSUE #15849: CompileStep.addRPath always generates absolute paths
### Zig Version

0.11.0-dev.3295+7cb2e653a

### Steps to Reproduce and Observed Behavior

1. Start a new project using `zig init-exe`
2. Make a slight modification to build.zig:

```zig
pub fn build(b: *std.Build) void {
    const target = b.standardTargetOptions(.{});
    const optimize = b.standardOptimizeOption(.{});

    const exe = b.addExecutable(.{
        .name = "xx",
        .root_source_file = .{ .path = "src/main.zig" },
        .target = target,
        .optimize = optimize,
    });
    b.installArtifact(exe);

    // link to any installed library, doesn't matter
    exe.linkSystemLibrary("ssl");

    // specify a relative rpath
    exe.addRPath("lib");

    const run_cmd = b.addRunArtifact(exe);
    const run_step = b.step("run", "Run the app");
    run_step.dependOn(&run_cmd.step);
}
```

3. Build the project using `zig build`
4. Inspect the run path:
```
$ readelf --dynamic  zig-out/bin/test | grep RUNPATH
 0x000000000000001d (RUNPATH)            Library runpath: [/home/user/code/test/lib:/usr/lib/x86_64-linux-gnu:/lib/x86_64-linux-gnu]
```

`build-exe` works as expected:
```
$ zig build-exe src/main.zig --name test -lssl -rpath lib
$ readelf --dynamic  test | grep RUNPATH
 0x000000000000001d (RUNPATH)            Library runpath: [lib:/usr/lib/x86_64-linux-gnu:/lib/x86_64-linux-gnu]
```

### Expected Behavior

The relative path should be preserved. build-exe and build.zig should be consistent with each other.

### PR DESCRIPTION
**Pull request by yujiri8**
Closes #15849.

### CODE REVIEW

File: lib/std/Build/Step/Compile.zig [hunk 1]
```diff
@@ -1815,7 +1815,9 @@ fn make(step: *Step, prog_node: *std.Progress.Node) !void {
             .generated => {},
         };
 
-        zig_args.appendAssumeCapacity(rpath.getPath2(b, step));
+        std.debug.print("path is {s}\n", .{rpath.getPathRelative(b, step)});
```
Comment (andrewrk) [inline lib/std/Build/Step/Compile.zig:1818]: stray debug print

**Review by andrewrk — CHANGES_REQUESTED** [review]
This will make rpaths relative to the build root (i.e. to the directory containing build.zig). Is that what is desired? I think it should instead be relative to the installation prefix, no?

**Comment by yujiri8** [comment]
:shrug: this PR's purpose was only to make `CompileStep.adRPath` consistent with the `-rpath` flag to `build-exe`. I have no idea what behavior is desirable here, but if relative paths are being interpreted relative to the wrong root, isn't that a separate issue?

File: lib/std/Build/Step/Compile.zig (hunk 1 above)
Comment (yujiri8) [reply lib/std/Build/Step/Compile.zig]: Sorry, fixed that.

**Comment by hazeycode** [comment]
> This will make rpaths relative to the build root (i.e. to the directory containing build.zig). Is that what is desired? I think it should instead be relative to the installation prefix, no?

Possibly not. There is also an issue resolving `$ORIGIN` on linux, I presume that this is what should be used if one wants to use a path relative to the installation dir? There was similar issue (resolving `@executable_path` on Darwin) that was solved by @kubkon here: https://github.com/ziglang/zig/pull/15061

**Comment by yujiri8** [comment]
Alright. Well, I'm not going to do any more work on this. Others feel free to add commits or close.

**Pull request closed without merging**